src/util.h

```diff
@@ -125,6 +131,32 @@ static SECP256K1_INLINE void *manual_alloc(void** prealloc_ptr, size_t alloc_siz
     return ret;
 }
 
+/* Cleanses memory to prevent leaking sensitive info. Won't be optimized out. */
+#define SECP256K1_CLEANSE(v) secp256k1_cleanse(&(v), sizeof(v))
+static SECP256K1_INLINE void secp256k1_cleanse(void *ptr, size_t len) {
+#if defined(_MSC_VER)
+    /* SecureZeroMemory is guaranteed not to be optimized out by MSVC. */
+    SecureZeroMemory(ptr, n);
+#elif defined(__GNUC__)
```

Comment:
I couldn't figure out when GCC started to support assembly but it's some version <= 2.95... So if you have such an old compiler, I guess you're on your own.

Reply:
GCC 2.95 is likely the earliest we'd would ever encounter someone wanting to support. (Haiku OS uses it still, I believe :) )